ibmcloud/SECURE_EXECUTION.md

```diff
@@ -0,0 +1,102 @@
+# Secure Execution Support for Peer Pods on IBM Cloud VPC
+
+IBM Cloud VPC already supports to create Hyper Protect Virtual Server on LinuxONE. And Confidential Computing is enabled by using the [IBM Secure Execution](https://www.ibm.com/docs/en/linux-on-systems?topic=execution-introduction) for Linux technology. For more information, please refer to [Confidential computing with LinuxONE](https://cloud.ibm.com/docs/vpc?topic=vpc-about-se)
+
+To support Secure Execution for Peer Pods on IBM Cloud VPC, we need build a Secure Execution enabled custom VM image for pod VMs, to replace the non-SE custom image. This document describes how to build a SE enabled custom image, based on the existing [image build script](./image/build.sh) for IBM Cloud VPC.
+
+> **Note**: In [the document](https://www.ibm.com/docs/en/linux-on-systems?topic=execution-secure-workload) describe the details about execution secure workload, you can go through the document to get a simplified view about how your workload is protected.
+
+## Set up a demo environment without Secure Execution
+
+Follow the [README.md](./README.md) to setup a demo environment on IBM Cloud VPC. We need create a **s390x** architecture based cluster, so please include the following two lines in the `terraform.tfvars` file:
+```
+instance_profile_name = "bz2-2x8"
+image_name = "ibm-ubuntu-18-04-1-minimal-s390x-3"
+```
+
+## Prepare worker node to build SE enabled image
+
+1. **Build genprotimg**
+
+`genprotimg` is used to build an encrypted boot record from a given kernel, initial RAM disk, parameters, and public host-key document.
+
+For Ubuntu 18.04, we can build it from the source on GitHub:
+```bash
+git clone https://github.com/ibm-s390-linux/s390-tools.git
+cd s390-tools/genprotimg/
+apt install -y libglib2.0-dev libssl-dev libcurl4-openssl-dev
+make install
+```
+
+2. **Download host key document from Resource Link**
+
+The host key must match the host system for which the image is prepared. Please refer to document [Obtaining a host key document from Resource Link](https://www.ibm.com/docs/en/linux-on-systems?topic=execution-obtain-host-key-document#lxse_obtain_hkd) for details.
+
+As a registered user, access the search page:
+```
+https://www.ibm.com/servers/resourcelink/hom03010.nsf/pages/HKDSearch?OpenDocument
+```
+
+Need input the **machine type** and **machine serial number**, which can be obtained from `/proc/sysinfo` of the s390x worker node. Check the values of **Type** and **Sequence Code**. Machine serial number would be the last 5 or 7 characters. Please refer to [this documentation](https://www.ibm.com/docs/en/linux-on-systems?topic=tasks-find-machine-serial) for details.
+
+When you obtain the host key document, please send it to the worker node host keys documents directory.
+
+> **NOTE**: You can download multiple different host keys and prepare only one SE enabled image for different host systems.
+
+## Build the SE enabled qcow2 image
```

Comment:
Could you combine the steps by just running `make push` here instead, which will do both. 
From my understanding the build fails when you try and run this due to the addition of `se-` within `build.sh` on line 42.
This means when it reaches the push.sh file it can no longer find the path because it has added an extra prefix that is not included in `IMAGE_NAME`. 
Is this a required prefix or could we instead specify in the exports something similar to: 
```
export CLOUD_PROVIDER=ibmcloud
export IMAGE_NAME=se-<An image name with arch included e.g. image-s390x>
export IBMCLOUD_COS_REGION=<the region your IKS cluster and zVSI is running on>
export IBMCLOUD_VPC_REGION=$IBMCLOUD_COS_REGION
export IBMCLOUD_VPC_NAME=<your generated vpc name from the terraform>
export IBMCLOUD_VPC_SUBNET_NAME=<the linked subnet name to the above vpc>
export IBMCLOUD_COS_SERVICE_ENDPOINT="https://s3.${IBMCLOUD_COS_REGION}.cloud-object-storage.appdomain.cloud"
export IBMCLOUD_COS_SERVICE_INSTANCE=<your cos instance>
export IBMCLOUD_COS_BUCKET=<your cos bucket in the region your cluster runs in>
export IBMCLOUD_API_KEY=<your ibmcloud apikey>
export IBMCLOUD_API_ENDPOINT="https://cloud.ibm.com"
export GOPATH=/root/go

CLOUD_PROVIDER=ibmcloud SE_BOOT=1 HOST_KEYS_DIR=/root/ make push
```

Reply:
Update the push.sh and verify.sh to support SE image.

Reply:
use `IMAGE_NAME ` remove `se-` hardcode string.